backend/src/repositories/AttemptRepository.ts

```diff
@@ -42,6 +43,50 @@ export default class AttemptRepository extends AbstractAttemptRepository {
         return attemptsIterator(uid, this.connectionManager.getReadonlyConnection());
     }
 
+    /**
+     * Returns the win rate data for a certain user. 
+     * @param userId the user whose win rate is to be calculated.
+     * @param difficulty optional difficulty to determine specific winrate.
+     * @param scenarioName optional scenario name for even more specific winrate.
+     * @returns A pair of the winrate and total attempts given the passed parameters.
+     */
+    getWinRate(userId: string, difficulty?: number, scenarioName?: string): {winrate: number, totalAttempts: number} {
+        const connection = this.connectionManager.getReadonlyConnection();
+
+        let queryTotal = `SELECT COUNT(*) FROM attempts a 
+            LEFT JOIN exercise e ON e.id = a.exercise_id 
+            WHERE a.user_id = ?`;
+        let queryCorrect = `SELECT COUNT(*) 
+                FROM attempts a 
+                LEFT JOIN exercise e ON e.id = a.exercise_id 
+                WHERE a.user_id = ? AND a.correct = 1`
+
+        let totalAttempts = 0;
+        let winrate = 0;
+
+        if(typeof difficulty !== 'undefined'){
+            queryTotal += ` AND e.difficulty = ?`;
+            queryCorrect += ` AND e.difficulty = ?`;
+            if (typeof scenarioName !== 'undefined'){
+                queryTotal += ` AND e.difficulty = ?`;
+                queryCorrect += ` AND e.difficulty = ?`;
+
+                totalAttempts = connection.query<number>(queryTotal, [userId, difficulty, scenarioName]);
```

Comment:
I think this returns an { count: number } instead of number. am i wrong?

Reply:
I just confirmed that this does indeed returns an object.

![image](https://github.com/user-attachments/assets/1eb77777-2bec-4777-a6fd-57bd654b4e6a)

So it returns an object `totalAttempts["COUNT(*)"]` which then equals 0. You can do `SELECT COUNT(*) AS count` and then get the value in typescript by 
```ts
totalAttempts = connection.query<{ count: number}>(...)[0].count;
```
or you can use which specifically gets one item:
```suggestion
            totalAttempts = this.queryOneT<number>(queryTotal, [userId, difficulty])!;
```